cpp/validate_harvested_records.cc

```diff
@@ -282,13 +357,11 @@ bool IsRecordValid(DbConnection * const db_connection, const MARC::Record &recor
         journal_name_and_info = journal_name_to_info_map->find(journal_name);
     }
 
-    if (journal_name_and_info->second.isInDatabase()) {
-        if (not RecordMeetsExpectations(record, journal_name_and_info->first, journal_name_and_info->second)) {
-            ++(*missed_expectation_count);
-            return false;
-        }
-    } else {
-        AnalyseNewJournalRecord(record, first_record, &journal_name_and_info->second);
+    if (not RecordMeetsExpectations(record, journal_name_and_info->first, general_info, journal_name_and_info->second)) {
```

Comment:
I think we can take "missed_expectation_count" out of the argument list and increment it near the call site for this function instead.

Reply:
Do you really mean "RecordMeetsExpectations" or do you mean "IsRecordValid"?

Reply:
done